pkg/catalog/catalog_test.go

```diff
@@ -92,45 +92,91 @@ func TestVersionExist(t *testing.T) {
 
 func TestList(t *testing.T) {
 	type args struct {
-		name      string
-		namespace string
-		tags      []string
+		name               string
+		namespace          string
+		tags               []string
+		currentNSImageName string
+		currentNamespace   string
+		currentNSImageTags []string
```

Comment:
what is this variable for?

Reply:
There are 2 calls to the same function(although different params) https://github.com/redhat-developer/odo/pull/618/files#diff-6025e9b620e9321c03c11bf29e5c4dbdR401 and https://github.com/redhat-developer/odo/pull/618/files#diff-6025e9b620e9321c03c11bf29e5c4dbdR391 and the function to mock the call isn't per parameter(as in return x when function called with y as parameter else return z) and the mocking is to api function not its wrapper that is immediately used in our func...

Hence, inorder for the mocking function to be able to return values from both namespaces this was the only way possible because the mocking is not per parameter